### PR TITLE
 Simplify merging enrich output

### PR DESCRIPTION
**Pull request by dnhatn**
The merge logic in MergePositionsOperator is excessively complex and lacks flexibility. It relies on the source operator emitting pages with ascending positions. Additionally, this merge logic introduced an unusual method, `appendAllValuesToCurrentPosition`, to the Block.Builder. We should replace this with a simpler and more flexible approach. This PR uses a mechanism similar to the grouping aggregation. In fact, it is very close to the values aggregation. Initially, I considered using the GroupingState from ValuesAggregator. However,  unlike in the values aggregation, we don't expect many multi-values in enrich. Hence, I introduced the new EnrichResultBuilders instead.

### CODE REVIEW

File: x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/BooleanBlock.java [hunk 1]
```diff
@@ -223,19 +223,6 @@ sealed interface Builder extends Block.Builder, BlockLoader.BooleanBuilder permi
...
-         * in this builder.
-         */
-        @Override
-        Builder appendAllValuesToCurrentPosition(Block block);
-
-        /**
-         * Appends the all values of the given block into a the current position
-         * in this builder.
-         */
-        Builder appendAllValuesToCurrentPosition(BooleanBlock block);
```
Comment (dnhatn) [inline x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/BooleanBlock.java:237]: I think we should remove this method; it's unusual.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/X-EnrichResultBuilder.java.st [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            for (int i = 0; i < totalPositions; i++) {
+                final var cell = i < cells.size() ? cells.get(i) : null;
+                if (cell == null) {
+                    builder.appendNull();
+                    continue;
+                }
+                if (cell.length > 1) {
+                    builder.beginPositionEntry();
+                }
+                // TODO: sort and dedup
```
Comment (dnhatn) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/X-EnrichResultBuilder.java.st:118]: This is for a follow-up.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/MergePositionsOperator.java [hunk 3]
```diff
@@ -44,21 +45,13 @@
...
-    private int filledPositions = 0;
-    private final boolean singleMode;
-    private final int positionCount;
     private final int positionChannel;
 
-    private final Block.Builder[] outputBuilders;
-    private final int[] mergingChannels;
-    private final ElementType[] mergingTypes;
-    private PositionBuilder positionBuilder = null;
+    private final EnrichResultBuilder[] builders;
```
Comment (dnhatn) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/MergePositionsOperator.java:50]: This class becomes much simpler.

**Comment by dnhatn** [comment]
@nik9000 I have benchmarked this change, along with two incoming improvements: one for reading values in batch, and another for leveraging the bytesref ordinal block. With these three changes, enrich in our benchmark is 11 times faster: from 330ms to 30ms.

```
|  50th percentile latency |    esql_stats_enrich_rates_fares | 332.198  |  31.7257  | -300.473 |  ms | -90.45% |
|  90th percentile latency |    esql_stats_enrich_rates_fares | 335.589  |  32.3748  | -303.214 |  ms | -90.35% |
| 100th percentile latency |    esql_stats_enrich_rates_fares | 340.448  |  35.3605  | -305.087 |  ms | -89.61% |
|  50th percentile latency |       esql_stats_enrich_payments | 330.927  |  29.0027  | -301.924 |  ms | -91.24% |
|  90th percentile latency |       esql_stats_enrich_payments | 335.693  |  29.5329  | -306.16  |  ms | -91.20% |
| 100th percentile latency |       esql_stats_enrich_payments | 349.575  |  31.7998  | -317.775 |  ms | -90.90% |
|  50th percentile latency | esql_stats_enrich_payments_fares | 362.12   |  31.1805  | -330.939 |  ms | -91.39% |
|  90th percentile latency | esql_stats_enrich_payments_fares | 364.841  |  31.6378  | -333.204 |  ms | -91.33% |
| 100th percentile latency | esql_stats_enrich_payments_fares | 368.572  |  32.9891  | -335.583 |  ms | -91.05% |
```

**Review by nik9000 — APPROVED** [review]
Oh that is nice!

**Comment by dnhatn** [comment]
@nik9000 Thanks for reviewing :)

**Pull request merged**